src/Nethermind/Nethermind.Network/P2P/NettyRlpStream.cs

```diff
@@ -97,8 +98,12 @@ public override int Position
 
         public override int Length => _buffer.ReadableBytes + (_buffer.ReaderIndex - _initialPosition);
 
-        public override bool HasBeenRead => _buffer.ReadableBytes > 0;
+        public override bool HasBeenRead => _buffer.ReadableBytes <= 0;
 
         protected override string Description => "|NettyRlpStream|description missing|";
+        public void Dispose()
```

Comment:
Are we sure we are disposing it with every use?

Reply:
I was under the impression Dispose is also called automatically?

Reply:
it turns out we dont need to call this after all